script.js

```diff
@@ -2,83 +2,110 @@
  * Section
  * ======================================================================== */
 
-var orders = [];
-
 document.addEventListener('DOMContentLoaded', function() {
 
-  // get orders from page
+      // get orders from page
 
-  chrome.tabs.executeScript(null, {
-    code: "\
-      var orders = [];\
-      if (document.querySelectorAll('h1')[0].innerText.toLowerCase().indexOf('istoric comenzi') != -1) {\
-        var tags = document.querySelectorAll('.history-item')[0].querySelectorAll('.content');\
-        for (var i = 0; i < tags.length; i++) {\
-          orders.push({\
-            name: tags[i].querySelector('h4').innerText.trim(),\
-            price: tags[i].querySelector('.price').innerText.trim(),\
-          });\
+    chrome.tabs.executeScript(null, {
+        code: "\
+      var orders = [], iterator, tags, dinerNameDOMElement, dinerName, priceValueDOMElement, priceValue;\
+      if (document.location.hash.startsWith('#history')) {\
+        tags = document.querySelectorAll('.history-item')[0].querySelectorAll('.content');\
```

Comment:
If there's no `'.history-item'` on the page, `('.history-item')[0].querySelectorAll()` will throw an error. The if / else needs to be rethought.